src/state/orderbook/hooks.ts

```diff
@@ -56,9 +64,19 @@ export function useOrderbookActionHandlers(): {
 
   const onResetUserPrice = useCallback(
     (price: number) => {
-      dispatch(resetUserPrice({ price }))
+      const initialAuction = asks[0]
+      const userOrder = { price: Number(price), volume: userOrderVolume }
+      let calculatedAuctionPrice
+      if (initialAuction) {
+        calculatedAuctionPrice = CalculatorClearingPrice.fromOrderbook(
+          bids,
+          initialAuction,
+          userOrder,
+        )
+      }
```

Comment:
Why are you doing the calculation in here and not in the `orderbook/reducer.ts`?
Wouldn't it be more performant to do it there?

Reply:
I had doubts about where to place this, but this way my criteria:
1. I have seen how in most of the reducers in the project they are only used for state assignment.
2. Decouple the business logic of the library `redux`. (I still can't figure out how it could be more effective to do it there,  maybe I'm missing something)

But in case you suggest to move it to the `reducers`, I suppose that it would be in the cases of `resetUserVolume` and `resetOrderbookData` so that it continues to comply with:
>  the updating behaviour should be the same as on the chart.